dotnet/src/Experimental/Assistants/Internal/Assistant.cs

```diff
@@ -74,61 +67,73 @@ internal sealed class Assistant : IAssistant
             await restContext.CreateAssistantModelAsync(assistantModel, cancellationToken).ConfigureAwait(false) ??
             throw new KernelException("Unexpected failure creating assistant: no result.");
 
-        return new Assistant(resultModel, chatService, restContext, plugins);
+        return new Assistant(resultModel, restContext, plugins);
     }
 
     /// <summary>
     /// Initializes a new instance of the <see cref="Assistant"/> class.
     /// </summary>
     internal Assistant(
         AssistantModel model,
-        OpenAIChatCompletion chatService,
         OpenAIRestContext restContext,
         IEnumerable<IKernelPlugin>? plugins = null)
     {
         this._model = model;
         this._restContext = restContext;
 
-        var services = new ServiceCollection();
-        services.AddSingleton<IChatCompletion>(chatService);
-        services.AddSingleton<ITextCompletion>(chatService);
-        this.Kernel = new Kernel(services.BuildServiceProvider(), plugins is not null ? new KernelPluginCollection(plugins) : null);
+        var builder =
+            new KernelBuilder()
+                .WithOpenAIChatCompletion(this._model.Model, this._restContext.ApiKey);
+
+        if (plugins is not null)
+        {
+            builder.ConfigurePlugins(
+                c =>
+                {
+                    foreach (var plugin in plugins)
+                    {
+                        c.Add(plugin);
+                    }
```

Comment:
```suggestion
                    c.AddRange(plugins);
```

Reply:
Anyway, took the approach of adding to Kernel directly.